src/transport/SecureSessionMgr.cpp

```diff
@@ -338,12 +326,15 @@ void SecureSessionMgr::HandleGroupMessageReceived(const PacketHeader & packetHea
     PeerConnectionState * state = mPeerConnections.FindPeerConnectionState(packetHeader.GetEncryptionKeyID(), nullptr);
     VerifyOrReturn(state != nullptr, ChipLogError(Inet, "Failed to find the peer connection state"));
 
-    OnMessageReceived(packetHeader, state->GetPeerAddress(), std::move(msgBuf));
+    OnMessageReceived(state->GetPeerAddress(), std::move(msgBuf));
```

Comment:
This does not look right.  The incoming buffer does NOT include the packet header, but we're going to try to parse it out of there again?

Presumably what should happen is that in `MessageCounterSyncMgr::ProcessPendingGroupMsgs` we should not consume the header and we should not pass it in here at all.  That should be folded into the first commit in this PR.

Please audit the other changes here carefully, then request review when all issues like this have been addressed.

Reply:
Done. I went through the messaging stack, and I didn't find any other case like this.  MCSP is the only component which need to resend the packet to SecureSessionMgr for re-processing.

Reply:
#5389 has removed all usage of `ProcessPendingGroupMsgs`. According to the spec, MCSP is only triggered by group message, but the packet processing procedure is exactly the same as normal messages, so there is no reason to add a separate code path for group message.

Here Yufeng's patch is fine for the context here. It helps me remove packetHeader in MCSP retrans table.

Reply:
@bzbarsky-apple PTAL?